jwst/pathloss/path_loss.py

```diff
@@ -344,6 +348,51 @@ def do_correction(input_model, pathloss_model):
         input_model.wavelength = wavelength_array
         input_model.meta.cal_step.pathloss = 'COMPLETE'
 
+    elif exp_type == 'NIS_SOSS':
+        """NIRISS SOSS pathloss correction is basically a correction for the
+        flux that falls outside the subarray aperture.  The correction depends
+        on the pupil wheel position and column number (or wavelength).  The
+        simpler option is to do the correction by column number, then the only
+        interpolation needed is a 1-d interpolation into the pupil wheel position
+        dimension."""
+
+        # Omit correction if this is a TSO observation
+        if input_model.meta.visit.tsovisit:
+            log.warning("NIRISS_SOSS TSO observations skip the pathloss step")
```

Comment:
No underscore in NIRISS_SOSS (an underscore is only used in the EXP_TYPE value "NIS_SOSS")

Reply:
OK